### PR TITLE
Run unit tests for full dotnet 4.6.2

### PR DESCRIPTION
**Pull request by wojtpl2**
I used `System.CommandLine` in [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet) (https://github.com/dotnet/BenchmarkDotNet/issues/1016)
but after my changes unit tests in BenchmarkDotNet for the classic .NET 4.x failed.

The problem is in the `RootCommand()` counstructor. It used `Assembly.GetEntryAssembly()`. 
https://github.com/dotnet/command-line-api/blob/cfc24749f3f3f5483a4946a912ff04fbdf574449/src/System.CommandLine/RootCommand.cs#L36-L37
`Assembly.GetEntryAssembly()` doesn't work in unit test for the classic .NET 4.x. There is an issue: https://github.com/Microsoft/vstest/issues/649

In this PR I used `(Assembly.GetEntryAssembly() ?? Assembly.GetExecutingAssembly());` instead of `Assembly.GetEntryAssembly()`. Behaviour should not be changed during normal use.

### CODE REVIEW

**Comment by wojtpl2** [comment]
I can't see details of build. E. g.
image
Probably build error is caused by the lack of .NET 4.7.2.
For Linux I can change the `build.ps1` script to skip running test for the classic .net. But I need help in case of Windows.
What version of full .net can I use?

**Comment by jonsequitur** [comment]
Here's the error:

/mnt/j/w/Private/dotnet_command-line-api/master/Release_Ubuntu16.04_prtest/artifacts/.dotnet/2.1.400/sdk/2.1.400/Microsoft.Common.CurrentVersion.targets(1179,5): error MSB3644: The reference assemblies for framework ".NETFramework,Version=v4.7.2" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [/mnt/j/w/Private/dotnet_command-line-api/master/Release_Ubuntu16.04_prtest/src/System.CommandLine.Tests/System.CommandLine.Tests.csproj]

File: src/System.CommandLine.Tests/System.CommandLine.Tests.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFrameworks>netcoreapp2.1;net472</TargetFrameworks>
```
Comment (adamsitnik) [inline src/System.CommandLine.Tests/System.CommandLine.Tests.csproj]: @wojtpl2 you can use an MSBuild property to make net472 target Windows-only

```suggestion
    <TargetFrameworks>netcoreapp2.1</TargetFrameworks>
    <TargetFrameworks Condition="'$(OS)' == 'Windows_NT'">$(TargetFrameworks);net472</TargetFrameworks>
```

**Comment by wojtpl2** [comment]
@jonsequitur I need information about version of .NET on Windows_NT platforms. I've tryed 4.7.2, 4.6.2. 4.6.1. 
I can't see details of build on Windows_NT.

**Comment by adamsitnik** [comment]
I have no idea why the CI machines are missing .NET SDK:

```log
D:\j\w\Debug_Windows---724fb6da\artifacts\.dotnet\2.1.400\sdk\2.1.400\Microsoft.Common.CurrentVersion.targets(1179,5): error MSB3644: The reference assemblies for framework ".NETFramework,Version=v4.6.2" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [D:\j\w\Debug_Windows---724fb6da\src\System.CommandLine.Tests\System.CommandLine.Tests.csproj]
```

Could you please try with `net471` before we get any response from the Team based in Redmond?

**Comment by adamsitnik** [comment]
@wojtpl2 it did not help, sorry for misleading you

```log
error MSB3644: The reference assemblies for framework ".NETFramework,Version=v4.7.1" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [D:\j\w\Debug_Windows---724fb6da\src\System.CommandLine.Tests\System.CommandLine.Tests.csproj]
```

**Comment by wojtpl2** [comment]
@adamsitnik No problem. I look at build script from clr https://github.com/dotnet/coreclr/blob/master/eng/common/tools.ps1#L184 
Maybe something is not configured.

**Comment by adamsitnik** [comment]
@wojtpl2 please give us some time, @jonsequitur is working internally at MS to solve the lack of SDK problem

**Comment by jonsequitur** [comment]
@dotnet test this please

**Comment by jonsequitur** [comment]
@wojtpl2 We've moved our build to Arcade. You should now be able to see the build results and I believe the original issue is resolved, if you'd like to rebase and retry.

I created a copy of your branch here: https://dev.azure.com/dnceng/public/_build/results?buildId=115329. There's a different error (Windows only) but I'm guessing it's related to the repo-wide configuration of the xunit version. I'm looking into it.

**Comment by wojtpl2** [comment]
Thanks for the information. Now everything is passed.

**Pull request merged**